mlrun/model_monitoring/controller.py

```diff
@@ -41,173 +39,174 @@
-                return max(
-                    self._first_request,
-                    self._stop - first_period_in_seconds,
-                )
-            return self._first_request
+    def _update_db(self, schedules: _Schedules) -> None:
+        self._db.put(json.dumps(schedules))
 
+    def _init_last_analyzed(self) -> int:
         logger.info(
-            "Got the last analyzed time for this endpoint and application",
-            endpoint=self._endpoint,
+            "No last analyzed time was found for this endpoint and application, as this is "
+            "probably the first time this application is running. Initializing last analyzed "
+            "to the latest between first request time or last update time minus one day",
             application=self._application,
-            last_analyzed=last_analyzed,
+            first_request=self._first_request,
+            last_updated=self._stop,
+        )
+        # max between one day and the base period
+        first_period_in_seconds = max(_SECONDS_IN_DAY, self._step)
+        return max(
+            self._first_request,
+            self._stop - first_period_in_seconds,
         )
-        return last_analyzed
+
+    def _get_schedules_and_last_analyzed(self) -> tuple[_Schedules, int]:
+        try:
+            content = self._db.get(encoding=_UTF_8)
+            try:
+                schedules = json.loads(content)
+                if self._application in schedules:
+                    return _Schedules(schedules), schedules[self._application]
+                else:
+                    last_analyzed = self._init_last_analyzed()
+                    schedules.update(_Schedules({self._application: last_analyzed}))
+            except json.JSONDecodeError:
+                # Using the earliest safe time to avoid TSDB misorders
+                last_analyzed = self._stop
+                schedules = _Schedules({self._application: last_analyzed})
+                logger.warning(
+                    "The monitoring schedules file is corrupted, resetting it "
+                    "with the last request as last_analyzed",
+                    path=self._db.url,
+                    content=content,
+                    last_analyzed=last_analyzed,
+                )
+        except (
+            mlrun.errors.MLRunNotFoundError,  # V3ioStore, InMemoryStore
+            FileNotFoundError,  # FileStore
+            BotoClientError,  # S3Store
+        ) as err:
+            if isinstance(err, BotoClientError):
+                # accept only "NoSuchKey" errors codes - equivalent to `FileNotFoundError`
+                if err.response["Error"]["Code"] != "NoSuchKey":
+                    raise
+            last_analyzed = self._init_last_analyzed()
+            schedules = _Schedules({self._application: last_analyzed})
+            logger.info(
+                "The monitoring schedules file does not exist for this endpoint, creating it",
+                path=self._db.url,
```

Comment:
Please add the creation of this json to be part of the model endpoint creation, and also we need to delete it when calling to `delete_model_endpoints_resources`.